consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/PandaPrinter.java

```diff
@@ -31,18 +34,48 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class PandaPrinter implements InSyncListener, ForkchoiceMessageListener, MergeStateHandler {
+public class PandaPrinter implements InSyncListener, ForkchoiceMessageListener, BlockAddedObserver {
+
+  private static PandaPrinter INSTANCE;
+
+  public static PandaPrinter init(final Optional<Difficulty> currentTotal, final Difficulty ttd) {
+    if (INSTANCE != null) {
+      LOG.debug("overwriting already initialized panda printer");
+    }
+
+    INSTANCE = new PandaPrinter(currentTotal, ttd);
+
+    return INSTANCE;
+  }
+
+  public static PandaPrinter getInstance() {
+    if (INSTANCE == null) {
+      throw new IllegalStateException("Uninitialized, unknown ttd");
+    }
+    return INSTANCE;
+  }
+
+  public PandaPrinter(final Optional<Difficulty> currentTotal, final Difficulty ttd) {
```

Comment:
since this is a singleton, should this be private?

Reply:
Yes it should! But since our tests are multi-threaded and singletons become a problem, I made that public to simplify testing.

Reply:
I see, but it could be package private